### PR TITLE
More details for test status in avocado vt-runner

### PR DESCRIPTION
**Pull request by richtja**
This will add the `fail_reason` and `exit_code` attributes to the
`FinishedMessage` inside vt-runner. Thanks to that, the user will have
more details about test status.

Reference: #3107
Signed-off-by: Jan Richter <jarichte@redhat.com>

### CODE REVIEW

**Comment by richtja** [comment]
Hi @pevogam this is a fix related to your https://github.com/avocado-framework/avocado-vt/pull/3201#issuecomment-912659714, please have a look.

**Comment by pevogam** [comment]
I got a bit delayed today but will take a look first thing tomorrow.

**Comment by pevogam** [comment]
Unfortunately I still don't see any details being displayed. @richtja have you tested this with an arbitrary VT test? It might help if you have some of the VT environment around for changes to the VT runner in general.

**Comment by richtja** [comment]
Hi @pevogam, can you please give me some reproducible, becase when I use your example with 
```
raise exceptions.TestSkipError("There is a known problem in ABC")
```

I get
```
/home/jarichte/Avocado/avocado-vt/avocado_vt/plugins/vt_resolver.py:40: UserWarning: The VT NextRunner is experimental and doesn't have current Avocado VT features
  warnings.warn("The VT NextRunner is experimental and doesn't have "
JOB ID     : b8f1cfd157e2b65869bdf13326be1a0175087c18
JOB LOG    : /home/jarichte/avocado/job-results/job-2021-09-08T11.04-b8f1cfd/job.log
 (1/1) io-github-autotest-qemu.boot: STARTED
 (1/1) io-github-autotest-qemu.boot:  SKIP: There is a known problem in ABC
RESULTS    : PASS 0 | ERROR 0 | FAIL 0 | SKIP 1 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/jarichte/avocado/job-results/job-2021-09-08T11.04-b8f1cfd/results.html
JOB TIME   : 21.47 s
```

**Comment by pevogam** [comment]
Hi @richtja,

Indeed, a SKIP in particular seems to work:
```
 (1-mis1mf331/9) normal.nongui.directory.usersync.external_users.smbserver: SKIP: There is a known problem in ABC
```
However, what I tested was with an ERROR and there were no details in that case. Haven't tested for FAIL but we have to make sure the current approach works for all status types: FAIL, ERROR, WARN, SKIP are the ones I know of and use often.

**Comment by richtja** [comment]
> Hi @richtja,
> 
> Indeed, a SKIP in particular seems to work:
> 
> ```
>  (1-mis1mf331/9) normal.nongui.directory.usersync.external_users.smbserver: SKIP: There is a known problem in ABC
> ```
> 
> However, what I tested was with an ERROR and there were no details in that case. Haven't tested for FAIL but we have to make sure the current approach works for all status types: FAIL, ERROR, WARN, SKIP are the ones I know of and use often.

Ok now I understand, there is a problem with ERROR status, I will look at that, thanks

**Comment by richtja** [comment]
@pevogam I did a force-push with fix

**Review by pevogam — COMMENTED** [review]
Hi @richtja, the error status works now. However, looking at the differences from the last push raises a few questions.

File: avocado_vt/plugins/vt_runner.py [hunk 1]
```diff
@@ -43,21 +43,26 @@ def params(self):
...
+            status_info = exc_info[1]
+            status = status_info.status
+            status_details = status_info.args
             if status == "ERROR" or status not in teststatus.STATUSES:
                 status = "ERROR"
                 self.queue.put(messages.StderrMessage.get(traceback.format_exc()))
         except Exception:
             status = "ERROR"
+            exc_info = sys.exc_info()
+            status_details = exc_info[1].args
```
[the remark this answers is not in the document]
Comment (richtja) [reply avocado_vt/plugins/vt_runner.py:62]: Actually, the `TestError` class be a subclass of the `TestBaseException`. The fix which I did in force-push was to include details for any generic exception. `TestError` was already working with details.

[the remark this answers is not in the document]
Comment (pevogam) [reply avocado_vt/plugins/vt_runner.py:62]: I see, I guess I should have tested with intentional `raise exceptions.TestError` in my case to spot the difference.

**Review by pevogam — APPROVED** [review]
Alright, this PR indeed delivers the additional details needed for the different statuses.

**Review by willianrampazzo — APPROVED** [review]

**Comment by pevogam** [comment]
Hmmm, I might be seeing a problem where some errors completely drop the status now:
```
 (1a1a1a1-mao1/25) all.original.install.mao1.qemu_kvm_android.default_bios.no_9p_export.smallpages.no_pci_assignable.qcow2.ide.smp2.rtl8139.Linux.Android.Oreo.official: STARTED
 (1a1a1a1-mao1/25) all.original.install.mao1.qemu_kvm_android.default_bios.no_9p_export.smallpages.no_pci_assignable.qcow2.ide.smp2.rtl8139.Linux.Android.Oreo.official: PASS (113.31 s)
 (1a1a1-mao1/25) all.internal.automated.customize.mao1.qemu_kvm_android.default_bios.no_9p_export.smallpages.no_pci_assignable.qcow2.ide.smp2.rtl8139.Linux.Android.Oreo.official: STARTED
 (1a1-mao1/25) all.internal.automated.on_customize.mao1.qemu_kvm_android.default_bios.no_9p_export.smallpages.no_pci_assignable.qcow2.ide.smp2.rtl8139.Linux.Android.Oreo.official: STARTED
 (1a1-mao1/25) all.internal.automated.on_customize.mao1.qemu_kvm_android.default_bios.no_9p_export.smallpages.no_pci_assignable.qcow2.ide.smp2.rtl8139.Linux.Android.Oreo.official: SKIP: Snapshot 'customize' of net1/mao1/image1 doesn't exist. Aborting due to passive mode.
```
I reproduced this completely with a simple 
```
from avocado.utils import process
...

def run(test, params, env):
    process.run("test 1 = 2")
```
VT test. The status report line is missing completely. @richtja could you try this with a simple VT test and tell me if you can reproduce it too?

**Comment by richtja** [comment]
Hi @pevogam yes I can reproduce it, thanks for reporting.

**Comment by pevogam** [comment]
Alright, this handles the same situation better and is even a bit more compact implementation. :heavy_check_mark:

**Pull request merged**